### PR TITLE
Resolve all ignored template lint errors

### PR DESCRIPTION
**Pull request by bertdeblock**
Noticed these when doing #717. Thought I'd might as well fix these in a separate PR.

### CODE REVIEW

File: app/templates/components/main-layout.hbs [hunk 1]
```diff
@@ -9,7 +9,6 @@
 
     {{#each @sortedGroupedResults as |result|}}
       <h2 class="anchorable-toc rounded-lg padding-small">Deprecations Added in {{result.since}} </h2>
-      {{renderId}}
```
Comment (bertdeblock) [inline app/templates/components/main-layout.hbs:12]: I'm guessing this was a leftover from testing something? Couldn't find any occurrence of `renderId` in the entire codebase.

File: app/templates/components/search-input.hbs [hunk 2]
```diff
@@ -3,7 +3,6 @@
   <input
     id="search-input"
     type="search"
-    value={{value}}
```
Comment (bertdeblock) [inline app/templates/components/search-input.hbs:6]: `value` was never used / set somewhere else.

File: app/templates/components/search-result.hbs [hunk 3]
```diff
@@ -12,11 +12,11 @@
               {{#if index}}
                 >
               {{/if}}
-              {{{heading.value}}}
+              {{html-safe heading.value}}
```
Comment (bertdeblock) [inline app/templates/components/search-result.hbs:15]: Not sure how you guys feel about adding new dependencies. Let me know if this should be changed.

File: app/templates/components/toc-section.hbs [hunk 4]
```diff
@@ -1,9 +1,10 @@
 <button
   id={{this.id}}
   class="cp-Panel-toggle level-1 list-unstyled"
-  onclick={{action (mut this.open) (not this.open)}}
+  onclick={{fn (mut this.open) (not this.open)}}
```
Comment (bertdeblock) [inline app/templates/components/toc-section.hbs:4]: `not` is used, but it seems `ember-truth-helpers` is not defined as a dependency.

**Review by jaredgalanis — APPROVED** [review]
This is a very nice cleanup, thank you! 👍

File: app/templates/components/search-input.hbs (hunk 2 above)
Comment (jaredgalanis) [reply app/templates/components/search-input.hbs:6]: good catch.

File: app/templates/components/search-result.hbs (hunk 3 above)
Comment (jaredgalanis) [reply app/templates/components/search-result.hbs:15]: I'm okay with this.

File: app/templates/components/toc-section.hbs (hunk 4 above)
Comment (jaredgalanis) [reply app/templates/components/toc-section.hbs:4]: @bertdeblock it seems to be functioning as expected. We can maybe take it out in follow up.

Comment (bertdeblock) [reply app/templates/components/toc-section.hbs:4]: @jaredgalanis Indeed, it's pulled in by `ember-styleguide`. I can create a new PR to add the dependency? Just to make sure we don't depend on `ember-styleguide` for `ember-truth-helpers` to be included.

**Pull request merged**